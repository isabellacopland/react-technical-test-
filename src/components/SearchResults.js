import React from "react";
import PropTypes from "prop-types";
import "../styles/searchresults.css";

const SearchResults = ({ results }) => {
    if (!results.length) {
        return <p>No results</p>
    } else {
        return (
            <div className="search-results__image">
            {results.map((image) => (
            <img 
            className="card-image"
            src={image}
            alt="spaceImage" />
            ))}
            </div>
        );
    }
};

export default SearchResults; 

SearchResults.propTypes = {
    results: PropTypes.arrayOf(PropTypes.string)
};