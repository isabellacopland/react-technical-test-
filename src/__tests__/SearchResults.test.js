import React, { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search Results", () => {
    const validProps = {
        results: ["test", "test"],
    };

  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults results={validProps.results}/>);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders placeholder it no results", () => {
      const validProps = [];
      render(<SearchResults results={validProps} />);

      const placeholderMsg = screen.getByText("No results");

      expect(placeholderMsg).toBeInTheDocument();
    });
});