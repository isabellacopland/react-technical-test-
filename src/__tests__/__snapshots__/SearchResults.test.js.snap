// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results__image"
  >
    <img
      alt="spaceImage"
      class="card-image"
      src="test"
    />
    <img
      alt="spaceImage"
      class="card-image"
      src="test"
    />
  </div>
</DocumentFragment>
`;
